dsfs/dataset.go

```diff
@@ -320,11 +320,16 @@ func setEntryCount(ds *dataset.Dataset, data cafs.File, mu *sync.Mutex, done cha
 	}
 
 	entries := 0
+	depth := 0
```

Comment:
Think we should start with 

`depth := 1   // original enclosure`

Reply:
for the record this resulted in a wonderful debate that was resolved by unit tests.